cassandra-migration/src/main/java/org/cognitor/cassandra/migration/Database.java

```diff
@@ -241,4 +297,73 @@ public Database setConsistencyLevel(ConsistencyLevel consistencyLevel) {
         this.consistencyLevel = notNull(consistencyLevel, "consistencyLevel");
         return this;
     }
+
+    /**
+     * Attempts to acquire the lead on a migration through a LightWeight
+     * Transaction.
+     *
+     * @param repositoryLatestVersion
+     *            the latest version number in the migration repository
+     * @return if taking the lead succeeded
+     */
+    boolean takeLeadOnMigrations(int repositoryLatestVersion) {
+        if (VersionNumber.parse("2.0").compareTo(cassandraVersion) > 0) {
+            // No LWT before Cassandra 2.0 so leader election can't happen
+            return true;
+        }
+
+        while (repositoryLatestVersion > getVersion()) {
+            try {
+                LOGGER.debug("Trying to take lead on schema migrations");
+                BoundStatement boundStatement = takeMigrationLeadStatement.bind(getKeyspaceName(), this.instanceId);
+                ResultSet lwtResult = session.execute(boundStatement);
+
+                if (lwtResult.wasApplied()) {
+                    LOGGER.debug("Took lead on schema migrations");
+                    tookLead = true;
+                    return true;
+                }
+
+                LOGGER.info("Schema migration is locked by another instance. Waiting for it to be released...");
+                waitFor(TAKE_LEAD_WAIT_TIME);
+            } catch (com.datastax.driver.core.exceptions.InvalidQueryException e1) {
+                // A little redundant but necessary
+                LOGGER.info("All required tables do not exist yet, waiting for them to be created...");
+                waitFor(TAKE_LEAD_WAIT_TIME);
+            }
+        }
+
+        return false;
+    }
+
+    private void waitFor(int waitTime) {
+        try {
+            Thread.sleep(waitTime);
+        } catch (InterruptedException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    /**
+     * Attempts to release the lead on schema migrations, if it was taken by the
+     * local process.
+     */
+    void removeLeadOnMigrations() {
+        if (tookLead) {
+            LOGGER.debug("Trying to take lead on schema migrations");
+
+            BoundStatement boundStatement = releaseMigrationLeadStatement.bind(getKeyspaceName(), this.instanceId);
+            ResultSet lwtResult = session.execute(boundStatement);
+
+            if (lwtResult.wasApplied()) {
+                LOGGER.debug("Released lead on schema migrations");
+                tookLead = false;
+                return;
+            }
+            // Another instance took the lead on migrations?
+            // Otherwise, TTL will do the trick
+            LOGGER.error("Could not release lead on schema migrations");
```

Comment:
Is this really an error or rather a warning? To me an error is something that requires immediate human intervention and might potentially page somebody out of bed. This seems to happen in case somebody else took the lead or the migration took longer than the TTL. I am not sure there is really much somebody can do about this except for investigate it. I think I would drop this to LogLevel.WARN.

Reply:
Right, I'll downgrade to `WARN`.